### PR TITLE
Kc34251/setup review

### PR DESCRIPTION
**Pull request by kc34251**
## Changes:
- Changed location and name of script_config file to paths, added color formatting, fixed overall code format. 

## Considerations:
- Needs a final review

### CODE REVIEW

File: scripts/run/detect_display_errors.sh [hunk 1]
```diff
@@ -7,29 +7,30 @@
...
     last_error_timestamp=0
 else
-    initial_errors=$(dmesg | grep -c -E "$DISPLAY_ERROR_MESSAGE")
+    initial_errors=$(dmesg | grep -c -E "${DISPLAY_ERROR_MESSAGE}")
     error_count=$((initial_errors % error_count_threshold))
 fi
 
 dmesg --follow | while read line; do
-    if echo "$line" | grep -q -E "$DISPLAY_ERROR_MESSAGE"; then
+    if echo "$line" | grep -q -E "${DISPLAY_ERROR_MESSAGE}"; then
```
Comment (jbesoto) [inline scripts/run/detect_display_errors.sh]: Add curly braces for consistency.
```diff
- if echo "$line" | grep -q -E "${DISPLAY_ERROR_MESSAGE}"; then
+ if echo "${line}" | grep -q -E "${DISPLAY_ERROR_MESSAGE}"; then
```

File: scripts/run/init_display.sh [hunk 2]
```diff
@@ -28,8 +28,8 @@ init_display(){
 }
 
 # Check if DISPLAY_FRAME_BUFFER_PATH exists
-if [ -e "$DISPLAY_FRAME_BUFFER_PATH" ]; then
+if [ -e "${DISPLAY_FRAME_BUFFER_PATH}" ]; then
     init_display
 else
-    echo "WARNING: $DISPLAY_FRAME_BUFFER_PATH does not exist."
+    echo "WARNING: ${DISPLAY_FRAME_BUFFER_PATH} does not exist."
```
Comment (jbesoto) [inline scripts/run/init_display.sh]: Does not follow status message formatting.

File: scripts/setup/setup.sh
Comment (jbesoto) [inline scripts/setup/setup.sh:1]: These globals are not visible to child processes unless they are exported. For instance, 
```diff
+ export RED GREEN YELLOW BLUE MAGENTA CYAN WHITE BOLD RESET
```

File: scripts/setup/setup.sh [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+    BLUE=""
+    MAGENTA=""
+    CYAN=""
+    WHITE=""
+    BOLD=""
+    RESET=""
+fi
+
+# Status Messages
+OK_MSG="${GREEN}[OK]     ${RESET}"
```
Comment (jbesoto) [inline scripts/setup/setup.sh]: For consistency, lets change the name of these variables to:
```diff
- OK_MSG="${GREEN}[OK] ${RESET}"
- WARNING_MSG="${YELLOW}[WARNING]${RESET}"
- FAIL_MSG="${RED}[FAIL]   ${RESET}"
+ OK="${GREEN}[OK] ${RESET}"
+ WARNING="${YELLOW}[WARNING]${RESET}"
+ FAIL="${RED}[FAIL]   ${RESET}"
```
Fix all references

File: scripts/run/detect_display_errors.sh
Comment (jbesoto) [inline scripts/run/detect_display_errors.sh:1]: This file has places where variable substitution is not wrapped with curly braces. Please fix.

**Review by jbesoto — CHANGES_REQUESTED** [review]
Please fix minor issues.

File: scripts/run/detect_display_errors.sh [hunk 4]
```diff
@@ -9,28 +9,28 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
...
         new_error_timestamp=$(echo "$line" | grep -oP '\[\s*\K\d+\.\d+')
```
Comment (NxFerrara) [inline scripts/run/detect_display_errors.sh]: Add curly braces here.

File: scripts/run/detect_display_errors.sh [hunk 5]
```diff
@@ -9,28 +9,28 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
...
         if (( $(echo "$new_error_timestamp > $last_error_timestamp" | bc -l) )); then
```
Comment (NxFerrara) [inline scripts/run/detect_display_errors.sh]: Add curly braces here.

**Review by NxFerrara — CHANGES_REQUESTED** [review]
Overall great work. Minor fixes.

File: scripts/run/detect_display_errors.sh (hunk 4 above)
Comment (jbesoto) [reply scripts/run/detect_display_errors.sh:30]: Duplicate observation. Removing

**Review by NxFerrara — APPROVED** [review]
Looks good.

**Review by jbesoto — APPROVED** [review]
Good job.  Keep it up

**Pull request merged**